src/pcapi/scripts/beneficiary/delete_or_suspend_account_from_file.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import Iterable
+
+from pcapi.core.offerers.models import UserOfferer
+from pcapi.core.users import constants
+from pcapi.core.users.api import suspend_account
+from pcapi.core.users.models import Favorite
+from pcapi.core.users.models import User
+from pcapi.models import db
+from pcapi.models.beneficiary_import import BeneficiaryImport
+from pcapi.models.beneficiary_import_status import BeneficiaryImportStatus
+from pcapi.models.deposit import Deposit
+
+
+logger = logging.getLogger(__name__)
+
+
+def _find_users_to_suspend(user_ids: set) -> set:
+    return set(
+        user_id[0] for user_id in Deposit.query.filter(Deposit.userId.in_(user_ids)).with_entities(Deposit.userId)
+    )
+
+
+def _delete_users_and_favorites(user_ids: set) -> None:
+    for user_id in user_ids:
+        logger.info("Deleting user and their data", extra={"user": user_id})
+        Favorite.query.filter(Favorite.userId == user_id).delete()
+        BeneficiaryImportStatus.query.filter(
+            BeneficiaryImportStatus.beneficiaryImportId == BeneficiaryImport.id,
+            BeneficiaryImport.beneficiaryId == user_id,
+        ).delete(synchronize_session=False)
+        BeneficiaryImport.query.filter(BeneficiaryImport.beneficiaryId == user_id).delete(synchronize_session=False)
+        user_offerers: list[UserOfferer] = UserOfferer.query.filter(UserOfferer.userId == user_id).all()
+        for user_offerer in user_offerers:
+            if user_offerer.isValidated and user_offerer.offerer.isValidated:
+                raise Exception("Trying to delete a pro user with a valid offerer")
+            db.session.delete(user_offerer)
+        User.query.filter(User.id == user_id).delete()
+        db.session.commit()
+
+
+def _suspend_users(user_ids: set, admin_email_used: str) -> None:
+    admin = User.query.filter_by(email=admin_email_used, isAdmin=True).one()
+    for user_id in user_ids:
+        user = User.query.filter(User.id == user_id).first()
```

Comment:
```suggestion
        user = User.query.filter(User.id == user_id).get()
```

Je suis de la Brigade Anti-First ! Des gens m'appellent le relou. Je me demande bien pourquoi... ;)